### PR TITLE
Migrate squid sensor on new API

### PR DESCRIPTION
**Pull request by vilchik-elena**

### CODE REVIEW

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/coverage/PythonCoverageSensor.java [hunk 1]
```diff
@@ -98,108 +91,80 @@
...
+      zeroMeasuresWithoutReports(context, filesCoveredByUT, filesCoveredByIT, filesCoveredOverall, linesOfCode);
     }
   }
 
-  private void zeroMeasuresWithoutReports(SensorContext context,
-                                          HashSet filesCoveredByUT,
-                                          HashSet filesCoveredByIT,
-                                          HashSet filesCoveredOverall
-    ) {
+  private void zeroMeasuresWithoutReports(
```
Comment (SonarTech) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/coverage/PythonCoverageSensor.java]: MAJOR Make "zeroMeasuresWithoutReports" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/coverage/CoberturaParser.java [hunk 2]
```diff
@@ -49,48 +49,51 @@ public void stream(SMHierarchicCursor rootCursor) throws XMLStreamException {
...
+        NewCoverage coverage = coverageData.get(inputFile);
+        if (coverage == null) {
+          coverage = context.newCoverage().onFile(inputFile);
+          coverageData.put(inputFile, coverage);
+        }
+        collectFileData(classCursor, coverage);
+
+      } else {
+        LOG.debug("Cannot find the file '{}', ignoring coverage measures", fileName);
+        classCursor.getNext();
```
Comment (pynicolas) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/coverage/CoberturaParser.java]: @vilchik-elena Should we do that? Don't we skip en entry?

**Comment by pynicolas** [comment]
Looks good to me.

**Pull request merged**